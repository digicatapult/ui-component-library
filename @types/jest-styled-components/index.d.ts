declare module 'jest-styled-components/serializer'
