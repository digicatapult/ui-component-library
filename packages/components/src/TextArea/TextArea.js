import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'

import colors from '../colors'

const { errorRed, grey } = colors

const StyledTextArea = styled.textarea`
  border: 1px solid ${grey};
  padding: 10px;
  border-radius: 5px;
`

const TextArea = (props) => <StyledTextArea {...props} />

export default TextArea
