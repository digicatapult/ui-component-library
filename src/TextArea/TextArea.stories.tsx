import React from 'react'
import { TextArea } from '../index.js'
import { Story } from '@storybook/react'

export default {
  title: 'TextArea',
  component: TextArea,
  argTypes: {},
}

const Template = (args: any) => <TextArea {...args} />

export const Default: Story = Template.bind({})
Default.args = { label: 'label', id: 'default' }

export const ErrorMessage: Story = Template.bind({})
ErrorMessage.args = {
  label: 'label',
  id: 'error-message',
  error: 'required field',
}
