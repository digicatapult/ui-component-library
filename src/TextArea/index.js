import TextArea from './TextArea.js'
export default TextArea
