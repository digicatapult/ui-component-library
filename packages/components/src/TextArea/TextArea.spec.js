import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import TextArea from '.'

describe('TextArea', () => {
  test('default', () => {
    const tree = renderer.create(<TextArea label="label" id="test" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
