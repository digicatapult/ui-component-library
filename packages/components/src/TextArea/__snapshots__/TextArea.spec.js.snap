// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea default 1`] = `
.c0 {
  border: 1px solid #D1D1D1;
  padding: 10px;
  border-radius: 5px;
}

<textarea
  className="c0"
  id="test"
  label="label"
/>
`;
